docs/design/architecture.md

```diff
@@ -170,47 +397,89 @@ The following diagram illustrates the Kata Containers network hotplug workflow.
 ![Network Hotplug](arch-images/kata-containers-network-hotplug.png)
 
 ## Storage
-Container workloads are shared with the virtualized environment through [virtio-fs](https://virtio-fs.gitlab.io/).
 
-The [devicemapper `snapshotter`](https://github.com/containerd/containerd/tree/master/snapshots/devmapper) is a special case. The `snapshotter` uses dedicated block devices rather than formatted filesystems, and operates at the block level rather than the file level. This knowledge is used to directly use the underlying block device instead of the overlay file system for the container root file system. The block device maps to the top read-write layer for the overlay. This approach gives much better I/O performance compared to using `virtio-fs` to share the container file system.
+### virtio SCSI
+
+If a block-based graph driver is configured,
+`virtio-scsi` is used to "share" the workload image (such as
```

Comment:
Does it share the workload, copy it, or give the VM exclusive access to block device range?  I think you could go really specific here if you wanted since it is an arch document.  I think going into detail here is good because people will wonder if this is secure or not. If it is shared it doesn't sound too secure.